Tribler/Core/Modules/restapi/search_endpoint.py

```diff
@@ -31,73 +46,198 @@ def render_GET(self, request):
-                        "modified": 14598395,
-                        "can_edit": False
-                    }
+                   "torrents":[
+                      {
+                         "commit_status":1,
+                         "num_leechers":0,
+                         "date":"1539867830.0",
+                         "relevance_score":0,
+                         "id":21,
+                         "size":923795456,
+                         "category":"unknown",
+                         "public_key":"4c69624e...",
+                         "name":"ubuntu-18.10-live-server-amd64.iso",
+                         "last_tracker_check":0,
+                         "infohash":"8c4adbf9ebe66f1d804fb6a4fb9b74966c3ab609",
+                         "num_seeders":0,
+                         "type":"torrent"
+                      },
+                      ...
+                   ],
+                   "chant_dirty":false
                 }
+
         """
-        if 'q' not in request.args:
-            request.setResponseCode(http.BAD_REQUEST)
-            return json.dumps({"error": "query parameter missing"})
 
-        # Notify the events endpoint that we are starting a new search query
-        self.events_endpoint.start_new_query()
-
-        # We first search the local database for torrents and channels
-        query = unicode(request.args['q'][0], 'utf-8')
-        keywords = split_into_keywords(query)
-
-        results_local_channels = self.channel_db_handler.search_in_local_channels_db(query)
-        with db_session:
-            results_local_channels.extend(map(convert_channel_metadata_to_tuple,
-                                              self.session.lm.mds.ChannelMetadata.search_keyword(query)))
-
-        results_dict = {"keywords": keywords, "result_list": results_local_channels}
-        self.session.notifier.notify(SIGNAL_CHANNEL, SIGNAL_ON_SEARCH_RESULTS, None, results_dict)
-
-        torrent_db_columns = ['T.torrent_id', 'infohash', 'T.name', 'length', 'category',
-                              'num_seeders', 'num_leechers', 'last_tracker_check']
-        results_local_torrents = self.torrent_db_handler.search_in_local_torrents_db(query, keys=torrent_db_columns)
-        with db_session:
-            results_local_torrents.extend(map(convert_torrent_metadata_to_tuple,
-                                              self.session.lm.mds.TorrentMetadata.search_keyword(query)))
-        results_dict = {"keywords": keywords, "result_list": results_local_torrents}
-        self.session.notifier.notify(SIGNAL_TORRENT, SIGNAL_ON_SEARCH_RESULTS, None, results_dict)
-
-        # Create remote searches
-        try:
-            self.session.search_remote_torrents(keywords)
-            self.session.search_remote_channels(keywords)
-        except OperationNotEnabledByConfigurationException as exc:
-            self._logger.error(exc)
-
-        return json.dumps({"queried": True})
+        first = 0
```

Comment:
Memo: this needs to be refactored/split into multiple methods.

Reply:
Really? When the old database is gone, only a single screen of Pony text will remain there.

Reply:
As the old database will stay in for a few months still I think we need to, yes. As this is a collective effort, it doesn't need to be you personally who does this ofcourse.

Reply:
We are going to move old contents into Pony database via the migration script. This giant mess will be gone then.